Services/Administration/OJS.Services.Administration.Data/Implementations/ContestsDataService.cs

```diff
@@ -138,13 +138,16 @@ public Task<bool> IsUserInExamGroupByContestAndUser(int id, string? userId)
                     c.ExamGroups.Any(eg => eg.UsersInExamGroups.Any(u => u.UserId == userId)));
 
         private async Task<int> GetMaxPointsByIdAndProblemGroupsFilter(int id, Expression<Func<ProblemGroup, bool>> filter)
-            => await this.GetByIdQuery(id)
-                .Select(c => c.ProblemGroups
-                    .AsQueryable()
-                    .Where(pg => pg.Problems.Any(p => !p.IsDeleted))
-                    .Where(filter)
-                    .Sum(pg => (int?)pg.Problems.First().MaximumPoints))
-                .FirstOrDefaultAsync() ?? default(int);
+        {
+            var problemGroups = await this.GetByIdQuery(id)
+                .SelectMany(c => c.ProblemGroups)
+                .Where(pg => pg.Problems.Any(p => !p.IsDeleted))
+                .Where(filter)
+                .ToListAsync();
+
+            return problemGroups
+                .Sum(pg => pg.Problems.First().MaximumPoints);
```

Comment:
This is not an efficient way to get what you want here. Why retrieve the ProblemGroups, if all you need is the Problems (and it's just the Problem's MaximumPoints). On top of that, it seems like [N+1 query problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping), as for every ProblemGroup you will make extra db call to get the `.Problems` of it, because they are not `.Include()`d in the ToListAsync result.
Also the logic is again modified. What is what we want to get here?

Reply:
The main problem with the previous query was the following error: "Cannot perform an aggregate function on an expression containing an aggregate or a subquery.\nCannot perform an aggregate function on an expression containing an aggregate or a subquery.". 
For this specific case, I need the first problem maximum points of each problemGroup. Please let me know, if there is a better solution for the problem. @gogo4ds

Reply:
From what I understand, you need the maximum points a user can achieve in a contest?
You can try:
```
=> await this.GetByIdQuery(id)
                .SelectMany(c => c.ProblemGroups)
                .Where(filter)
                .Select(pg => pg.Problems
                    .Where(p => !p.IsDeleted)
                    .Select(p => p.MaximumPoints)
                    .FirstOrDefault())
                .SumAsync(maxPoints => maxPoints);
```

Reply:
Unfortunately, I have already tried this and got the following error: "Cannot perform an aggregate function on an expression containing an aggregate or a subquery."